File: 170320/04_yield/01.py
#!/usr/bin/env python3

# Пример корутин к одному из проектов по C++.

def foo():
   print("a1")
   yield 1  # "Приостанавливает" выполнение функции. Получаем функцию-генератор, которая генерирует элементы по очереди.
   print("a2")
   yield 2
   print("a3")
   yield 3
   print("a4")


def bar():
   while True:
       yield 1


n = foo()
print(next(n))
print(next(n))
print(next(n))
print(next(n))

File: 170417/02_casts/01.cpp
int main() {
  // Тут не нужен - и так произойдёт.
  int a = static_cast<int>(1e9), b = static_cast<int>(1e9);
  long long x = static_cast<long long>(a) * b;
  return 0;
}


class Foo {
};

class Bar : Foo {
};

Foo* x;
Bar* y;

x = y;  // Можно.
y = static_cast<Bar*>(x);  // Можно, но аккуратно.

static_cast<Bar>(*x);

// данные <FOO> данные
//        ^
